guide/restful_service/order_mgt_service.bal

```diff
@@ -76,25 +74,29 @@ service<http:Service> orderMgt bind listener {
         methods: ["POST"],
         path: "/order"
     }
-    addOrder(endpoint client, http:Request req) {
-        json orderReq = check req.getJsonPayload();
-        string orderId = orderReq.Order.ID.toString();
-        ordersMap[orderId] = orderReq;
-
-        // Create response message.
-        json payload = { status: "Order Created.", orderId: orderId };
-        http:Response response;
-        response.setJsonPayload(untaint payload);
-
-        // Set 201 Created status code in the response message.
-        response.statusCode = 201;
-        // Set 'Location' header in the response message.
-        // This can be used by the client to locate the newly added order.
-        response.setHeader("Location", "http://localhost:9090/ordermgt/order/" +
-                orderId);
-
-        // Send response to the client.
-        _ = client->respond(response);
+    resource function addOrder(http:Caller caller, http:Request req) {
+        var orderReq = req.getJsonPayload();
+        if (orderReq is json) {
+            string orderId = orderReq.Order.ID.toString();
+            ordersMap[orderId] = orderReq;
+
+            // Create response message.
+            json payload = { status: "Order Created.", orderId: orderId };
+            http:Response response = new;
+            response.setJsonPayload(untaint payload);
+
+            // Set 201 Created status code in the response message.
+            response.statusCode = 201;
+            // Set 'Location' header in the response message.
+            // This can be used by the client to locate the newly added order.
+            response.setHeader("Location", "http://localhost:9090/ordermgt/order/" +
+                    orderId);
+
+            // Send response to the client.
+            _ = caller->respond(response);
+        } else {
+            panic orderReq;
```

Comment:
Shall we send an error response rather than panicing?
```ballerina
    response.statusCode = 500;
    response.setPayload(untaint <string>orderReq.detail().message);
    _ = caller->respond(response);
```